backend/app/core/utils/querying_utils.py

```diff
@@ -1,9 +1,54 @@
-import bcrypt
+from sqlmodel import Session, select
+
+from app.core.exceptions import EmailAlreadyExistsError, UserCreationError
+from app.core.models.auth import Auth
+from app.core.models.role import Role, RoleType
+from app.core.models.session import Session as SessionModel
+from app.core.models.user import User
+from app.core.schemas.user_register import UserRegister
+from app.database import engine
 
 
 class QueryingUtils:
 	@staticmethod
-	def hash_password(password: str) -> str:
-		password_bytes = password.encode("utf-8")
-		hashed_bytes = bcrypt.hashpw(password_bytes, bcrypt.gensalt())
-		return hashed_bytes.decode("utf-8")
+	def get_default_role_id(session: Session) -> str:
+		default_role = session.exec(select(Role).where(Role.role == RoleType.USER)).first()
+		if not default_role:
+			raise ValueError("Default role not found")
+		return default_role.id
+
+	@staticmethod
+	def register(user: UserRegister, ip: str) -> str:
+		with Session(engine) as session:
+			if session.exec(select(Auth).where(Auth.email == user.email)).first():
+				raise EmailAlreadyExistsError
+
+			new_auth = Auth(
+				email=user.email,
+				password=user.password,
+				role_id=QueryingUtils.get_default_role_id(session),
+			)
+			session.add(new_auth)
+
+			new_user = User(
+				name=user.name,
+				last_name=user.last_name,
+				date_of_birth=user.date_of_birth,
+				auth_id=new_auth.id,
+			)
+			session.add(new_user)
+
+			searched_user = session.exec(select(User).where(User.auth_id == new_auth.id)).first()
```

Comment:
as I've said before we do not need to perform those checks here. If there is an error when running the code without `session.exec` use `session.flush()` instead